src/shared/start-gateway.js

```diff
@@ -29,6 +30,14 @@ export async function startGateway<
         process.env.GAE_SERVICE = name;
     }
 
+    /**
+     * Setup logging.
+     * We create the root logger once and then share it via a singleton.
+     * This avoids us creating a new one in each worker, which was happening
+     * when we created the logger on import of `getLogger`.
+     */
+    setRootLogger(logger);
+
```

Comment:
Instead of creating a logger implicitly in the `getLogger` call, we now set it here based off the options.